compiler/ballerina-lang/src/main/java/io/ballerina/projects/PackageResolution.java

```diff
@@ -197,6 +199,15 @@ PackageVersion getVersionFromPackageManifest(PackageOrg requestedPkgOrg, Package
         return null;
     }
 
+    String getRepositoryFromPackageManifest(PackageOrg requestedPkgOrg, PackageName requestedPkgName) {
```

Comment:
Can we update the `getVersionFromPackageManifest` method to return the dependency? Then we don't need to iterate dependencies twice.

Reply:
Fixed with b139f6d